components/Markets/ConditionalMarketOrderBook.tsx

```diff
@@ -145,9 +145,11 @@ export function ConditionalMarketOrderBook({
       } else {
         _spreadString = `${spread.toFixed(2).toString()} (${spreadPercent}%)`;
       }
-      setSpreadString(_spreadString);
+      setSpreadString(
+        (curSpreadString) => curSpreadString === _spreadString ? curSpreadString : _spreadString
+      );
 
-      setWsConnected(true);
+      setWsConnected((curConnected) => curConnected === false);
```

Comment:
why do we disconnect ?

Reply:
https://github.com/metaDAOproject/meta-dao-frontend/blob/064c005a13cd19adb79399cd1717b6a75c8e3fb7/components/Markets/ConditionalMarketOrderBook.tsx#L199C11-L199C33

So we disconnect on error with the subscription request. And only after receiving good data and updating do we set the connection to true.

This could be something that is incorrect. As well as you stated before this SHOULD exist above this component.

Reply:
And quite honestly we should have a master WS connection object which is leveraged instead of individual ones across the space.

Reply:
Will resolve for now, but otherwise this is a fair point.